### ISSUE TITLE
New linter: detect repeated literal spaces in regexes

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2972]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

Multiple adjacent literal spaces in a regular expression are difficult to count and are often accidental. A quantifier communicates the intended width more clearly.

```clojure
#"foo   bar"
```

**Describe the solution you'd like**

Add a `:regex-spaces` linter, enabled at warning level by default, that reports adjacent unescaped spaces outside character classes and suggests a quantifier. Do not warn in extended mode, inside character classes, or for escaped spaces.

**Describe alternatives you've considered**

A generic style linter could cover this, but regex parsing requires context for escapes, character classes, and flags. Simple string matching would create false positives.

**Additional context**

I am interested in submitting a focused PR with escape, character class, extended-mode, and language tests if this approach is accepted.

**Comment by NoahTheDuke** [comment]
Does this need to be a wholly separate linter? Would a `:regex` linter that covers multiple issues be better here?

**Comment by marco-carvalho** [comment]
An umbrella regex linter could avoid keyword proliferation. I separated this from `:invalid-regex` because invalid syntax is a correctness error while repeated spaces are a style warning, and users may reasonably configure them differently. The regression corpus found four repeated-space cases: two in ClojureScript sources and two in Metabase. If a broader `:regex` linter is preferred, it would need a configuration shape that preserves independent levels for syntax and style checks; nested per-check levels would be new compared with the usual clj-kondo linter configuration.

**Comment by yuhan0** [comment]
Agreed that this feels more like a stylistic concern and possibly out-of-scope for clj-kondo. What were the examples found by the regression corpus?
